File: xkcd/src/main/java/xyz/jienan/xkcd/comics/ComicsPagerAdapter.java
package xyz.jienan.xkcd.comics;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import xyz.jienan.xkcd.home.base.BaseStatePagerAdapter;
import xyz.jienan.xkcd.comics.fragment.SingleComicFragment;

public class ComicsPagerAdapter extends BaseStatePagerAdapter {


    public ComicsPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        SingleComicFragment fragment = SingleComicFragment.newInstance(position + 1);
        fragmentsMap.put(position + 1, fragment);
        return fragment;
    }
}

File: xkcd/src/main/java/xyz/jienan/xkcd/comics/ComicsPagerAdapter.kt
package xyz.jienan.xkcd.comics

import androidx.fragment.app.Fragment
import xyz.jienan.xkcd.comics.fragment.SingleComicFragment
import xyz.jienan.xkcd.home.base.BaseStatePagerAdapter

class ComicsPagerAdapter(fragment: Fragment) : BaseStatePagerAdapter(fragment) {

    override fun createFragment(position: Int)= SingleComicFragment.newInstance(position + 1)
}
